In [1]:
import numpy as np
import pandas as pd

In [2]:
coursera_df = pd.read_csv(r'rough_data\coursera_1.5m_reviews\coursera_reviews.csv')

In [3]:
coursera_df

reviews        reviewers  \
0        Pretty dry, but I was able to pass with just t...      By Robert S   
1        would be a better experience if the video and ...   By Gabriel E R   
2        Information was perfect! The program itself wa...       By Jacob D   
3        A few grammatical mistakes on test made me do ...        By Dale B   
4        Excellent course and the training provided was...        By Sean G   
...                                                    ...              ...   
1454706                                                  g     By Brijesh K   
1454707                                                  .    By Vasavi V M   
1454708                                                  .     By Drishti D   
1454709                                                  .  By FAUSTINE F K   
1454710                                                  G        By Reza S   

         date_reviews  rating                 course_id  
0        Feb 12, 2020       4  google-cbrs-cpi-training  
1        Sep 28, 2020       4  google-cbrs-cpi-training  
2        Apr 08, 2020       4  google-cbrs-cpi-training  
3        Feb 24, 2020       4  google-cbrs-cpi-training  
4        Jun 18, 2020       4  google-cbrs-cpi-training  
...               ...     ...                       ...  
1454706  Aug 25, 2020       5       computer-networking  
1454707  Jul 02, 2020       5       computer-networking  
1454708  Jun 20, 2020       5       computer-networking  
1454709  Jun 07, 2020       5       computer-networking  
1454710  Aug 12, 2018       5       computer-networking  

[1454711 rows x 5 columns]

In [5]:
coursera_df['course_id'].value_counts()

python                                        45218
machine-learning                              35895
neural-networks-deep-learning                 35750
technical-support-fundamentals                35319
python-data                                   33546
                                              ...  
entreprise-changement-climatique                  6
social-economic-networks                          6
curanderismo-plants                               6
intro-to-healthcare                               6
building-modern-python-applications-on-aws        3
Name: course_id, Length: 604, dtype: int64

604 Unique courses in the dataset with 1,454,711 total reviews.

In [8]:
coursera_df['course_id'].value_counts()[:590]

python                              45218
machine-learning                    35895
neural-networks-deep-learning       35750
technical-support-fundamentals      35319
python-data                         33546
                                    ...  
attention-models-in-nlp                49
project-execution                      42
healthcarejobs                         33
finance-fundamentals                   27
accounting-data-analytics-python       24
Name: course_id, Length: 590, dtype: int64

589 Courses that have 25+ reviews.

In [12]:
coursera_df['course_id'].value_counts()[:586]

python                            45218
machine-learning                  35895
neural-networks-deep-learning     35750
technical-support-fundamentals    35319
python-data                       33546
                                  ...  
project-planning-1                   60
introduction-to-data-analytics       54
community-organizing                 54
write-your-first-novel               52
attention-models-in-nlp              49
Name: course_id, Length: 586, dtype: int64

585 Courses that have 50+ reviews

In [15]:
coursera_df['course_id'].value_counts()[:570]
                                       

python                                                        45218
machine-learning                                              35895
neural-networks-deep-learning                                 35750
technical-support-fundamentals                                35319
python-data                                                   33546
                                                              ...  
mastering-final-cut-pro                                         109
antibiotic-stewardship                                          107
project-management                                              106
introduction-to-computers-and-office-productivity-software      105
enterprise-architecture                                          99
Name: course_id, Length: 570, dtype: int64

569 Courses that have 100+ reviews.

In [17]:
coursera_df['rating'].mean()

4.696649025132827

Average value for a rating is 4.7 / 5. Note: This is not per course.